LoRaEngine/LoraKeysManagerFacade/DeviceGetter.cs

```diff
@@ -84,9 +84,14 @@ public async Task<List<IoTHubDeviceInfo>> GetDeviceList(string devEUI, string ga
                 string cacheKey = devEUI + devNonce;
                 using (var deviceCache = new LoRaDeviceCache(this.cacheStore, devEUI, gatewayId, cacheKey))
                 {
+                    if (deviceCache.HasValue())
```

Comment:
Just to add context, that was a bug that caused the function to not return "dev nonce already used" if the loser would not get the lock.